Update the color palette and adapt it to all the components

ChengYanJin opened the pull request:
**Component**: all the components

**Description**:
- update the light/dark color palette 
- add the color palette in storybook
- update the all the components

**Design**:
Navbar:
![image](https://user-images.githubusercontent.com/18453133/74251303-3b5a7c00-4cec-11ea-8048-40e1b5952023.png)
![image](https://user-images.githubusercontent.com/18453133/74251380-5927e100-4cec-11ea-88ea-af123632d4bf.png)
storybook:
![image](https://user-images.githubusercontent.com/18453133/74743869-76ffc380-5261-11ea-8824-532e0aacb20e.png)

**Breaking Changes**:

If you want to use the components, you may update to our default color palette or define your own colors
```
const themes = [
  {
    name: "Light Theme",
    brand: {
      base: "#607080",
      primary: "#FAF9FB",
      primaryDark1: "#F7F6F9",
      primaryDark2: "#EDEAF0",
      secondary: "#037AFF",
      secondaryDark1: "#1C3D59",
      secondaryDark2: "#1C2E3F",
      success: "#006F62",
      healthy: "#25AC56",
      healthyLight: "#75FE63",
      info: "#027AFF",
      warning: "#FEFA51",
      danger: "#EF3340"
      critical: "#BE2543",
      background: "#ffffff",
      backgroundBluer: "#ECF4FF",
      textPrimary: "#313B44",
      textSecondary: "#8593A0",
      textTertiary: "#6C8398",
      borderLight: "#A5A5A5",
      border: "#A5A5A5"
    }
  },
  {
    name: "Dark Theme",
    brand: {
      base: "#6A7B92",
      primary: "#1D1D1F",
      primaryDark1: "#171718",
      primaryDark2: "#0A0A0A",
      secondary: "#037AFF",
      secondaryDark1: "#1C3D59",
      secondaryDark2: "#1C2E3F",
      success: "#006F62",
      healthy: "#25AC56",
      healthyLight: "#75FE63",
      info: "#027AFF",
      warning: "#FEFA51",
      danger: "#EF3340",
      critical: "#BE2543",
      background: "#121214",
      backgroundBluer: "#182A41",
      textPrimary: "#FFFFFF",
      textSecondary: "#A8B5C1",
      textTertiary: "#A7B6C3",
      borderLight: "#2C3137",
      border: "#A5A5A5"
    }
  }
];
```

---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #165

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

ChengYanJin committed:
config: update the light/dark theme

- there are some useless props like base, baseContrast1, backgroundContrast1, backgroundContrast2
  and border to be remove( those props are used by other components, so it caonnot be removed now)

ChengYanJin committed:
Navbar: update the Navbar component to adapt the new color palette

bert-e commented:
# Hello chengyanjin,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/core-ui/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @carlito-scality

* @nicolas2bert

* @rachedbenmustapha

* @chengyanjin

* @monpote

ChengYanJin committed:
dropdown: update the dropdown component to the display of navbar

- there is still something need to be done in Dropdown component to adapt the new color palette

ChengYanJin committed:
storybook: add light/dark color palette

ChengYanJin committed:
button/dropdown: update the button and dropdown components

- add two props (secondaryDark1 and secondaryDark2) in color palette for the outline button
- add primaryButton props for the primary button

ChengYanJin committed:
breadcrumb: update the breadcrumb with the new color palette

refs: #165

ChengYanJin committed:
table: update the table component with the new color palette

- set the background the head with background color
- set the background of each row with the primary color
- set the border of each row with border color

Refs: #165

ChengYanJin committed:
tabs: update the tabs with the new color palette

- remove the bottom border
- remove the selected bold font

Refs: #165

ChengYanJin committed:
sidebar: update the sidebar component with the new color palette

Refs: #165

ChengYanJin committed:
textarea: update the textarea component

Refs: #165

ChengYanJin committed:
checkbox: update the checkbox component

Refs: #165

ChengYanJin committed:
select: update the select component

- the background color should be primary
- the chosen item background color should be primaryDark2

Refs: #165

ChengYanJin committed:
input: update the input component

Refs: #165

ChengYanJin committed:
searchInput: update the searchInput component

Refs: #165

ChengYanJin committed:
loader: update the color palette for loader

Refs: #165

ChengYanJin committed:
notification: update to the new color palette

there are three situations for the notifications
- default: primary background & textPrimary font color
- warning variant: yellow background & black font color
- others variant: variant background & white font color

Refs: #165

ChengYanJin committed:
stepper: update the stepper component with the new color palette

Refs: #165

ChengYanJin committed:
toggle: update to the new color palette

Refs: #165

ChengYanJin committed:
modal: update to the new color palette

- set the primary color to the ModalContent
- set the primaryDark1 to the footer

Refs: #165

ChengYanJin committed:
tooltip: update to the new color palette

- the background color for tooltip should be primaryDark1 to have contrast with the primary

Refs: #165

ChengYanJin committed:
vegalite: update the vegalite to the new color palette

Refs: #165

ChengYanJin committed:
chip: update to the new color palette

- set the font color for warning varient blackLight

Refs: #165

ChengYanJin committed:
progressbar: update to the new color palette

- progressbar
- cloudprogressbar
- circulargressbar

Refs: #165

ChengYanJin committed:
collapsiblePanel: update to new color palette

ChengYanJin committed:
button: change the default button type from primaryButton to base

Refs: #165

ChengYanJin committed:
update the test for all the components

ChengYanJin committed:
banner: adapt the banner component to the new color palette

Refs: #165

ChengYanJin commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.0`




There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.0`




Please check the status of the associated issue None.

Goodbye chengyanjin.
